integration/hibernate-base/src/main/java/com/blazebit/persistence/impl/hibernate/HibernateJpaProvider.java

```diff
@@ -502,12 +502,19 @@ protected String getMappedBy(CollectionPersister persister) {
         }
 
         // In this case, the property might represent a formula
-        if (columnNames.length == 1 && columnNames[0] == null) {
+        boolean isFormula = columnNames.length == 1 && columnNames[0] == null;
+        boolean isSubselect = entityPersister.getTableName().contains("(");
```

Comment:
Besides storing the SQL query segments in the entity persisters, the true value of the `Subselect` annotation only seems to be passed around during metamodel generation, and is not really obtainable through the metamodel. It may need some rework in order to support the *SQL from clause replacement * later on. An alternative that I have considered is `entityPersister.getEntityType().getReturnedClass().isAnnotationPresent(Subselect.class)`. Suggestions are really welcome here!

Reply:
I've switched to `tables.length == 1 && tables[0] == null` because my previous assumption has been shown to be wrong by CI. I don't really like the approach of `entityPersister.getEntityType().getReturnedClass().isAnnotationPresent(Subselect.class)`, because it has all sorts of edge cases (such as a Subselect in an inheritance hierarchy - not that I am aware that this would be allowed).